nupic/research/frameworks/pytorch/models/resnets.py

```diff
@@ -125,6 +112,12 @@ def conv_layer(
         padding=padding,
         bias=bias,
     )
+
+    weights_density = layer_params.weights_density
+    if layer_params.params_function is not None:
+        weights_density = layer_params.params_function(in_planes,
+                                                       out_planes,
+                                                       kernel_size).weights_density
```

Comment:
This looks confusing. Maybe `LayerParams` should be implemented as a class using `auto_sparse_params`  as the `__init__` instead of `namedtuple` with a parameter override function

Reply:
Yes, I agree. I was trying to stick to the existing namedtuple approach, but it would be a lot clearer as a class.

Reply:
Another option is to treat `auto_sparse_params` as a factory and remove the `params_function` from the namedtuple

Reply:
We still want an easy way to just specify the parameters and explore several different strategies. I think a class is cleaner - we can better hide the complexity, and experiment with different strategies through subclasses.

Reply:
agreed